File: demo/src/main/java/com/example/demo/entity/SPDV.java
package com.example.demo.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "SPDV")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SPDV {

    @Id
    @Column(name = "MaSPDV", length = 10)
    private String maSPDV;

    @NotNull
    @Size(max = 100)
    @Column(name = "TenSPDV", length = 100, nullable = false)
    private String tenSPDV;

    @NotNull
    @Column(name = "DonGia", nullable = false)
    private BigDecimal donGia;

    @Size(max = 10)
    @Column(name = "DVT", length = 10)
    private String dvt;

    @Size(max = 200)
    @Column(name = "MoTa", length = 200)
    private String moTa;

    @OneToMany(mappedBy = "spdv", cascade = CascadeType.ALL)
    private List<BangGia> bangGias = new ArrayList<>();

    @OneToMany(mappedBy = "spdv", cascade = CascadeType.ALL)
    private List<DinhMucCK> dinhMucCKs = new ArrayList<>();

    @OneToMany(mappedBy = "spdv", cascade = CascadeType.ALL)
    private List<CTHoaDon> ctHoaDons = new ArrayList<>();

    @OneToMany(mappedBy = "spdv", cascade = CascadeType.ALL)
    private List<CTPhieu> ctPhieus = new ArrayList<>();
}

File: back-end/app/models/spdv.py
from sqlalchemy import Column, String, Numeric, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime

from app.db.base_class import Base


class SPDV(Base):
    """Sản phẩm dịch vụ model."""
    
    # Loại bỏ id từ Base
    id = None
    
    __tablename__ = "spdv"
    
    maspdv = Column(String(10), primary_key=True, index=True)
    tenspdv = Column(String(100), nullable=False)
    dongia = Column(Numeric(18, 2), nullable=False)
    dvt = Column(String(10), nullable=False)
    mota = Column(String(200), nullable=True)
    
    # Relationships
    ct_hoa_don = relationship("CTHoaDon", back_populates="san_pham")
    ct_phieu = relationship("CTPhieu", back_populates="san_pham")